tools/version/src/main/java/org/apache/polaris/version/PolarisVersion.java

```diff
@@ -0,0 +1,245 @@
+   * -Prelease} project property.
+   *
+   * <p>Example value: {@code 21.0.5}
+   *
+   * @see #isReleaseBuild()
+   */
+  public static Optional<String> getBuildSystem() {
+    return PolarisVersionJarInfo.buildInfo(MF_BUILD_SYSTEM);
+  }
+
+  /**
+   * Returns the build timestamp as in the jar manifest, if {@linkplain #isReleaseBuild() build-time
+   * Git information} is available, when Polaris has been built with the Gradle {@code -Prelease}
+   * project property.
+   *
+   * <p>Example value: {@code 2024-12-16-11:54:05+01:00}
+   *
+   * @see #isReleaseBuild()
+   */
+  public static Optional<String> getBuildTimestamp() {
+    return PolarisVersionJarInfo.buildInfo(MF_BUILD_TIMESTAMP);
+  }
+
+  public static String readNoticeFile() {
+    return readResource("NOTICE");
+  }
+
+  public static String readSourceLicenseFile() {
+    return readResource("LICENSE");
+  }
+
+  public static String readBinaryLicenseFile() {
+    return readResource("LICENSE-BINARY-DIST");
+  }
+
+  private static final String MF_VERSION = "Apache-Polaris-Version";
+  private static final String MF_IS_RELEASE = "Apache-Polaris-Is-Release";
+  private static final String MF_BUILD_GIT_HEAD = "Apache-Polaris-Build-Git-Head";
+  private static final String MF_BUILD_GIT_DESCRIBE = "Apache-Polaris-Build-Git-Describe";
+  private static final String MF_BUILD_TIMESTAMP = "Apache-Polaris-Build-Timestamp";
+  private static final String MF_BUILD_SYSTEM = "Apache-Polaris-Build-System";
+  private static final String MF_BUILD_JAVA_VERSION = "Apache-Polaris-Build-Java-Version";
+  private static final List<String> MF_ALL =
+      List.of(
+          MF_VERSION,
+          MF_IS_RELEASE,
+          MF_BUILD_GIT_HEAD,
+          MF_BUILD_GIT_DESCRIBE,
+          MF_BUILD_TIMESTAMP,
+          MF_BUILD_SYSTEM,
+          MF_BUILD_JAVA_VERSION);
+
+  static String readResource(String resource) {
+    var fullResource = format("/META-INF/resources/apache-polaris/%s.txt", resource);
+    var resourceUrl =
+        requireNonNull(
+            PolarisVersion.class.getResource(fullResource),
+            "Resource " + fullResource + " does not exist");
+    try (InputStream in = resourceUrl.openConnection().getInputStream()) {
+      return new String(in.readAllBytes(), UTF_8);
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to load resource " + fullResource + " resource", e);
+    }
+  }
+
+  // Couple inner classes to leverage Java's class loading mechanism as singletons and for
+  // initialization. Package-protected for testing.
+
+  static final class PolarisVersionJarInfo {
+    static final Map<String, String> BUILD_INFO = new HashMap<>();
+
+    static Optional<String> buildInfo(String key) {
+      return Optional.ofNullable(BUILD_INFO.get(key));
+    }
+
+    static {
+      loadManifest("MANIFEST.MF");
+    }
+
+    static void loadManifest(String manifestFile) {
```

Comment:
Using a HashMap is probably safe here but the below variant makes it clear that the map is read-only:

```suggestion
    static {
      BUILD_INFO = Map.of(loadManifest("MANIFEST.MF"));
    }

    static Map<String, String> loadManifest(String manifestFile) {
```

Reply:
Sure, but that would break the fake-manifest test